feat: add support for lists of basic python types

Maistho opened the pull request:
Adds support for lists of strings/numbers etc

Let me know if there should be some test changes or such. I haven't contributed to this repository before, so I'd appreciate some pointers if so. :)

dbanty commented:
Awesome, thanks for the contribution 😃!  We do try to keep 100% unit test coverage as well as add end to end tests where feasible. General info should be in the CONTRIBUTING.md (though I just realized the layout is a bit misleading). 

In this case since your changes are in `openapi_python_client/parser/responses.py` the unit tests should be in `tests/test_parser/test_responses.py`.  You'll want to cover the new branch in `response_from_data` with a new function in `TestResponseFromData`. You'll notice the tests all start with `test_<name_of_function>` which is repetitive and annoying but it helps PyCharm locate tests 😁.

You'll also want to cover the new `ListBasicResponse` class with a `TestListBasicResponse` which you can probably do by just copying `TestBasicResponse` and tweaking as needed.

For the end to end tests, right now it's all based on a FastAPI app. The easiest thing to do would be to create a new endpoint which returns a basic list response, then run `task oge` to regenerate the "golden-master" which is a generated client. I realize basing this on FastAPI is probably not the most accessible thing, so if you want to leave that out I can add it in for you.

Finally, stick a summary of your addition in a bullet in CHANGELOG.md, feel free to include a `thanks @Maistho` in there to credit yourself 😄.

Maistho commented:
Hi, thanks for the detailed response. I hadn't the time last week but I'll make sure to write the tests and documentation changes required next week.

FastAPI is fine, that's what we use for our projects anyway.

Maistho commented:
I've implemented the unit and e2e tests, and added a note in the CHANGELOG.md now.

I also rebased on the latest changes on the `main` branch.

I guess there shouldn't be any other changes, such as documentation required?

dbanty commented on end_to_end_tests/golden-master/my_test_api_client/api/tests.py:
```diff
@@ -50,6 +50,66 @@ def get_user_list(
+    response = httpx.get(url=url, headers=headers,)
+
+    if response.status_code == 200:
+        return [str(item) for item in cast(List[str], response.json())]
```
We probably don't need the `cast` here since we're converting to the correct type with `str`. I suppose it doesn't hurt though, so we can leave it in.
